app/services/SubmissionService.scala

```diff
@@ -83,14 +83,28 @@ class SubmissionService @Inject()(val cacheConnector: DataCacheConnector,
       _ <- saveResponse(credId, subscription, SubscriptionResponse.key)
       _ <- enrol(safeId, subscription.amlsRefNo, request.businessDetailsSection.fold("")(_.registeredOffice match {
         case Some(o: RegisteredOfficeUK) => o.postCode
-        case _ => ""
+        case _ => getPostcodeFromTP(request)
       }), groupId, credId)
     } yield subscription) recoverWith {
       case e: Upstream4xxResponse if e.upstreamResponseCode == UNPROCESSABLE_ENTITY =>
         Future.failed(SubscriptionErrorResponse.from(e).fold[Throwable](e)(r => DuplicateSubscriptionException(r.message)))
     }
   }
 
+  private def getPostcodeFromTP(subscriptionRequest: SubscriptionRequest) = {
+
+    subscriptionRequest.tradingPremisesSection match {
+      case Some(tpSection) => {
+        tpSection.head.yourTradingPremises.map(x => x.tradingPremisesAddress) match {
```

Comment:
here instead of Some(tpSection) you could use 
Some(firstTp +: _) => 
or 
Some(Seq(firstTp, _ *)) =>
just to avoid using tpSection.head which can have similar effect to using get on option

Reply:
@slavvco - I think the above is difficult to read. the user can't submit with zero trading premises so using head should be fine. If it raises an exception then good as we have an incomplete section.